src/components/RBAC/RBAC.tsx

```diff
@@ -0,0 +1,30 @@
+import { MY_USER_DETAILS } from '@/features/user/schema/user'
+import NotAuthorized from '@/pages/NotAuthorized'
+import { useQuery } from '@apollo/client'
+import { useRouter } from 'next/router'
+import React, { PropsWithChildren, ReactElement } from 'react'
+
+interface IProps {
+  allowedRoles?: string[]
+}
+
+const RBAC = ({ children, allowedRoles }: PropsWithChildren<IProps>) => {
```

Comment:
Components should follow the `PascalCase` naming format, and maybe it can be better termed as `Authorization` rather than `RBAC`.